recipe-server/client/control/state/approvalRequests/actions.js

```diff
@@ -0,0 +1,97 @@
+import { getRequest } from './selectors';
+
+import {
+  APPROVAL_REQUEST_FETCH,
+  APPROVAL_REQUEST_FETCH_FAILURE,
+  APPROVAL_REQUEST_FETCH_SUCCESS,
+  APPROVAL_REQUEST_RECEIVE,
+  APPROVAL_REQUESTS_FETCH,
+  APPROVAL_REQUESTS_FETCH_FAILURE,
+  APPROVAL_REQUESTS_FETCH_SUCCESS,
+} from '../action-types';
+
+import apiFetch from '../../utils/apiFetch';
+
+
+function fetchApprovalRequestSuccess(dispatch, requestId, approvalRequest) {
+  dispatch({
+    type: APPROVAL_REQUEST_FETCH_SUCCESS,
+    requestId,
+  });
+
+  dispatch({
+    type: APPROVAL_REQUEST_RECEIVE,
+    approvalRequest,
+  });
+}
+
+
+function fetchApprovalRequestFailure(dispatch, requestId, error) {
+  dispatch({
+    type: APPROVAL_REQUEST_FETCH_FAILURE,
+    error,
+    requestId,
+  });
+}
+
+
+export function fetchApprovalRequest(pk) {
+  return (dispatch, getState) => {
+    const requestId = `fetch-${pk}`;
+    const request = getRequest(getState(), requestId);
+
+    if (request.loading) { return true; }
+
+    dispatch({
+      type: APPROVAL_REQUEST_FETCH,
+      requestId,
+    });
+
+    return apiFetch(`approval_request/${pk}/`, { method: 'GET' })
+      .then(approvalRequest => fetchApprovalRequestSuccess(dispatch, requestId, approvalRequest))
+      .catch(error => fetchApprovalRequestFailure(dispatch, requestId, error));
+  };
+}
+
+
+function fetchApprovalRequestsSuccess(dispatch, requestId, approvalRequests) {
+  dispatch({
+    type: APPROVAL_REQUESTS_FETCH_SUCCESS,
+    requestId,
+  });
+
+  approvalRequests.forEach(approvalRequest => {
+    dispatch({
+      type: APPROVAL_REQUEST_RECEIVE,
+      approvalRequest,
+    });
+  });
+}
+
+
+function fetchApprovalRequestsFailure(dispatch, requestId, error) {
+  dispatch({
+    type: APPROVAL_REQUESTS_FETCH_FAILURE,
+    error,
+    requestId,
+  });
+}
+
+
+export function fetchApprovalRequests() {
```

Comment:
Same issue I had with another function - we have `fetchApprovalRequest` and `fetchApprovalRequests`, which are nearly identical. Maybe it's just me, but I find functions with such similar names to be a little uncomfortable.

In general, I'm not sure if I'm totally on board with all of the separated `request/requests` logic. It's not super DRY, and in fact a little confusing, since the code is pretty much the same (except for a few `s`'s).

Couldn't we just set up the actions to handle single requests just the same as multiple? I don't think I see the benefit to handling singular vs multiple responses differently.